c_src/dpiData_nif.c

```diff
@@ -390,6 +390,18 @@ DPI_NIF_FUN(data_get)
         dataRet = enif_make_resource(env, stmtRes);
     }
     break;
+    case DPI_NATIVE_TYPE_ROWID:
+    {
+        ErlNifBinary bin;
+        dpiRowid* rowid = data->value.asRowid;
+        const char* string;
+        uint32_t stringlen; 
+        dpiRowid_getStringValue(rowid, &string, &stringlen);
```

Comment:
add `dpiContext *context` to `dpiDataPtr_res` and copy context in all places where `dpiDataPtr_res` are allocated/created, then use it in line above for error handling.